deployment/articles-dxp/01-saml/06-configuring-saml.markdown

```diff
@@ -0,0 +1,131 @@
+---
+header-id: configuring-saml
+---
+
+# Configuring SAML
+
+There are some ways of configuring the SAML plugin outside the UI. This is done
+via OSGi configuration files and by uploading metadata XML to configure how
+connections are negotiated. 
+
+## OSGi Configuration Properties
+
+As noted in the previous tutorials, anything related to configuring SP
+connections must be done through the SAML Admin UI where configurations are
+saved to Liferay's database. SP connections can no longer be made via properties
+files as they were in versions prior to 3.1.0. 
+
+| **Note:** Don't use OSGi `.config` files or @product@'s System Settings Control
+| Panel application to configure SAML providers (IdP or SP). The System Settings
+| UI is auto-generated, and is for advanced administrators. It does not perform the
+| enhanced validation on the fields that the SAML Admin UI performs, so it could
+| allow administrators to create invalid configurations.
+
+This is a portal instance scoped configuration which can be managed via OSGi
+Configuration Admin. The affected properties are those in the
+`SAMLProviderConfiguration` metatype:
+
+    - `saml.keystore.credential.password`
+    - `saml.sp.assertion.signature.required`
+    - `saml.idp.authn.request.signature.required`
+    - `saml.sp.clock.skew`
+    - `saml.default.assertion.lifetime`
+    - `saml.sp.default.idp.entity.id`
+	- `saml.enabled`
+	- `saml.entity.id`
+	- `saml.sp.ldap.import.enabled`
+	- `saml.role`
+	- `saml.idp.session.maximum.age`
+    - `saml.idp.session.timeout`
+    - `saml.sp.sign.authn.request`
+    - `saml.sign.metadata`
+    - `saml.ssl.required`
+    - `saml.idp.metadata.name.id.attribute`
+
+The SAML Admin UI remains the place for creating the portal instance scoped
+configuration instances.
+
+Note that there is also a system wide configuration, represented by the
+`SamlConfiguration` metatype. 
+
+If you used Liferay 6.2, please note that the following system wide properties
+were removed:
+
+    `saml.metadata.paths` (served no purpose after removal of SP connection defaults)
+    `saml.runtime.metadata.max.refresh.delay`
+    `saml.runtime.metadata.min.refresh.delay`
+
+The latter two properties were replaced with the single property
+`saml.runtime.metadata.refresh.interval`.
```

Comment:
@sez11a 

Change: there is no such property: we have `com.liferay.saml.runtime.configuration.SamlConfiguration.getMetadataRefreshInterval()`

Reply:
Okay.